modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteProject.java

```diff
@@ -135,7 +134,7 @@ public IgniteProject(RelInput input) {
         if (fieldCollations.isEmpty())
             return Pair.of(nodeTraits, List.of(in.replace(RelCollations.EMPTY)));
 
-        Map<Integer, Integer> targets = new HashMap<>();
+        Int2IntOpenHashMap targets = new Int2IntOpenHashMap();
```

Comment:
`Integer newIndex = targets.get(inFieldCollation.getFieldIndex());` below boxes value.
Seems, value can be null.